modules/core/shared/src/main/scala/weaver/internals/ExpectSame.scala

```diff
@@ -45,5 +25,21 @@ private[weaver] trait ExpectSame {
       found: A)(
       implicit eqA: Eq[A] = Eq.fromUniversalEquals[A],
       showA: Show[A] = Show.fromToString[A],
-      loc: SourceLocation): Expectations = eql(expected, found)
+      loc: SourceLocation): Expectations = eql(expected, found)(eqA, showA, loc)
+
+  def diff[A](
```

Comment:
What do you think about adding a new function vs modifying the existing ones?

Adding a new function to the API (`diff` or a different name) is the simplest approach to compatibility.

We could modify the signatures of the existing `expect` and `same` functions to replace `Eq` and `Show` with `Comparison`, but it would be difficult to keep the same behaviour. They currently use default implicit parameters to infer the `Show` and `Eq` traits.

The closest we could get is:

```scala
object Comparison {
  implicit def fromEqAndToString(
      implicit val eq: Eq[A],
      show[A]: Show[A] = Show.fromToString): Comparison[A] = ???
}
def eql[A](
    expected: A,
    found: A)(
    implicit comparison: Comparison[A], // finds Comparison.fromEqAndToString
    loc: SourceLocation): Expectations = ???

def same[A](
    expected: A,
    found: A)(
    implicit comparison: Comparison[A] = Comparison.fromUniversalEqAndToString,
    loc: SourceLocation): Expectations = ???
```

but `same` wouldn't respect a custom `Show` instance.

Reply:
Repeating the discord convo for posterity : I'm fine the suggestion, despite it breaking bincompat and behaviour on this.